engine/runtime-integration-tests/src/test/java/org/enso/interpreter/test/PrivateAccessTest.java

```diff
@@ -0,0 +1,113 @@
+package org.enso.interpreter.test;
+
+import static org.hamcrest.CoreMatchers.containsString;
+import static org.hamcrest.CoreMatchers.is;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.junit.Assert.fail;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import org.enso.polyglot.PolyglotContext;
+import org.enso.polyglot.RuntimeOptions;
+import org.graalvm.polyglot.PolyglotException;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.TemporaryFolder;
+
+public class PrivateAccessTest extends TestBase {
+  @Rule public TemporaryFolder tempFolder = new TemporaryFolder();
+
+  /**
+   * Private constructor defined in a type inside unknown module that has no project cannot be
+   * called at all, as it is impossible to determine the project name. Unknown modules are engine
+   * internal constructs that should not use private econstructors at all.
+   */
+  @Test
+  public void privateConstructorCannotBeCalledInUnknownProject() {
+    var src =
+        """
+        type My_Type
+            private Cons data
+        main = My_Type.Cons 42
+        """;
+    try (var ctx = createDefaultContext()) {
+      // TestBase.eval module creates a temporary source - an unknown module without a project.
```

Comment:
This is a bit surprising. Even it is an _unknown module without a project_, functions like `main` defined in the same module should be allowed to access `private` constructors defined in the same module.

Reply:
This was fixed by 47da50e553d38ddc356d3143ff0129bd066ad7bd. I have updated the test - now it should succeed and return value 42. The update of test is in 03b5199237eed55448082770cc62821c6d8e643f